src/app/(sidebar)/endpoints/[[...pages]]/page.tsx

```diff
@@ -423,15 +436,103 @@ export default function Endpoints() {
     }, delay);
   };
 
+  const getRpcPostPayloadProps = (endpoint: string) => {
+    const defaultRpcRequestBody = {
+      jsonrpc: "2.0",
+      id: 8675309,
+      method: pageData?.rpcMethod,
+    };
+
+    switch (endpoint) {
+      case Routes.ENDPOINTS_GET_EVENTS: {
+        const filteredParams = params.filters ? JSON.parse(params.filters) : {};
+
+        return {
+          ...defaultRpcRequestBody,
+          params: {
+            startLedger: params.ledger ?? "",
+            cursor: params.cursor,
+            limit: params.limit,
+            filters: [
+              {
+                type: filteredParams.type ?? "",
+                contractIds: filteredParams.contract_ids ?? "",
+                topics: filteredParams.topics ?? "",
+              },
+            ],
+          },
+        };
+      }
+      case Routes.ENDPOINTS_GET_LEDGER_ENTRIES: {
+        return {
+          ...defaultRpcRequestBody,
+          params: {
+            keys: params.transaction ?? "",
+          },
+        };
+      }
+      case Routes.ENDPOINTS_GET_TRANSACTION: {
+        return {
+          ...defaultRpcRequestBody,
+          params: {
+            hash: params.transaction ?? "",
+          },
+        };
+      }
+      case Routes.ENDPOINTS_GET_TRANSACTIONS: {
+        return {
+          ...defaultRpcRequestBody,
+          params: {
+            startLedger: params.ledger ?? "",
+            cursor: params.cursor,
+            limit: params.limit,
+          },
+        };
+      }
+      case Routes.ENDPOINTS_SEND_TRANSACTION: {
+        return {
+          ...defaultRpcRequestBody,
+          params: {
+            transaction: params.transaction ?? "",
+          },
+        };
+      }
+      case Routes.ENDPOINTS_SIMULATE_TRANSACTION: {
+        return {
+          ...defaultRpcRequestBody,
+          params: {
+            transaction: params.transaction ?? "",
+            resourceConfig: {
+              instructionLeeway: params.resourceConfig,
+            },
+          },
+        };
+      }
+      default: {
+        return defaultRpcRequestBody;
+      }
+    }
+  };
+
   const renderPostPayload = () => {
     if (pageData?.requestMethod === "POST") {
+      let renderedProps;
+
+      if (pathname === Routes.ENDPOINTS_TRANSACTIONS_POST) {
+        renderedProps = { tx: params.tx ?? "" };
+      }
+
+      if (IS_RPC_ENDPOINT) {
+        renderedProps = getRpcPostPayloadProps(pathname);
+      }
+
       return (
         <div className="Endpoints__txTextarea">
           <Textarea
             id="tx"
             fieldSize="md"
             label="Payload"
-            value={JSON.stringify({ tx: params.tx ?? "" }, null, 2)}
+            value={JSON.stringify(renderedProps, null, 2)}
```

Comment:
This will throw an error if `renderedProps` is undefined. We can either check if `renderedProps` is not undefined here or set the default value to an empty object.